api_tests/nodes/views/test_node_list.py

```diff
@@ -346,6 +376,31 @@ def test_incorrect_filtering_field_not_logged_in(self):
         assert_equal(len(errors), 1)
         assert_equal(errors[0]['detail'], "'notafield' is not a valid field for this endpoint.")
 
+    def test_filtering_on_root(self):
+        # Build up a family of nodes
+        node_structure = [5, 2, [1, 2], 3, [10, 2], 2, [4, [2, 2], 6]]
+        render_generations_from_node_structure_list(self.project_one, node_structure)
+        # create some unrelated projects
+        ProjectFactory(title="Road Dogg Jesse James", is_public=True)
+        ProjectFactory(title="Badd *** Billy Gunn", is_public=True)
+
+        url = '/{}nodes/?filter[root]={}'.format(API_BASE, self.project_one._id)
+
+        res = self.app.get(url, auth=self.user_one.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['links']['meta']['total'], 42)
```

Comment:
When possible, try not to rely on counts in tests, as they are very brittle (if someone adds `ProjectFactory()` in the wrong place, the test will fail in a sad way).

Better to make assertions against values that are more stable, such as primary keys.

```python
# make root project...
# make child of root project...
# make filtered request
# child should be included in the result
result_ids = [each['id'] for each in ....]
assert_in(child._id, result_ids)